### ISSUE TITLE
spec: Revise content of `spec/core/data_structures.md`

### ISSUE THREAD

**Issue by cason** [cometbft/cometbft#2211]
The spec file https://github.com/cometbft/cometbft/blob/main/spec/core/data_structures.md:

    Here we describe the data structures in the CometBFT blockchain and the rules for validating them.

However, when adapting it with PBTS parameters (#2206), we realized that possibly some descriptions and probably some validation rules are outdated or not matching the implementation.

It will be important to revise this content, as it is considered by developers and applications using CometBFT types.

NOTE: some updates were already performed in branch `feature/pbts`. Consider that as base branch or wait until it is merged to `main`.

**Comment by cason** [comment]
It is also a question whether this should be under  `spec`, as it looks more like a reference documentation to me.

**Comment by andynog** [comment]
I agree this is more like a reference documentation and would fit well under the new reference section in the docs. There might be some bits of information on the doc that are more spec related but I believe it should be fine to be under reference. When you merge the `feature/pbts` and include the content for that we can move this doc.
